scipy/stats/tests/test_stats.py

```diff
@@ -3723,123 +3677,158 @@ def do(self, a, b, axis=None, dtype=None):
+        assert_equal(x.dtype, dtype)
+
+class TestGeoMean(GeoMeanTestMethod):
+    def test_1d_list(self):
         #  Test a 1d list
         a = [10, 20, 30, 40, 50, 60, 70, 80, 90, 100]
         b = 45.2872868812
-        self.do(a, b)
+        self.equal_test(a, b)
 
-    def test_1darray(self):
+        a = [1, 2, 3, 4]
+        desired = power(1 * 2 * 3 * 4, 1. / 4.)
+        self.equal_test(a, desired, decimal=14)
+
+        desired1 = stats.gmean(a, axis=-1)
+        self.equal_test(a, desired1, axis=-1)
+
+    def test_1d_array(self):
         #  Test a 1d array
         a = np.array([10, 20, 30, 40, 50, 60, 70, 80, 90, 100])
         b = 45.2872868812
-        self.do(a, b)
+        self.equal_test(a, b)
 
-    def test_1dma(self):
+        a = array([1, 2, 3, 4], float32)
+        desired = power(1 * 2 * 3 * 4, 1. / 4.)
+        self.equal_test(a, desired, dtype=float32)
+
+        desired1 = stats.gmean(a, axis=-1)
+        self.equal_test(a, desired1, axis=-1, dtype=float32)
+
+    def test_1d_ma(self):
         #  Test a 1d masked array
         a = np.ma.array([10, 20, 30, 40, 50, 60, 70, 80, 90, 100])
         b = 45.2872868812
-        self.do(a, b)
+        self.equal_test(a, b)
 
-    def test_1dmavalue(self):
+    def test_1d_ma_value(self):
         #  Test a 1d masked array with a masked value
         a = np.ma.array([10, 20, 30, 40, 50, 60, 70, 80, 90, 100], mask=[0,0,0,0,0,0,0,0,0,1])
         b = 41.4716627439
-        self.do(a, b)
+        self.equal_test(a, b)
 
     # Note the next tests use axis=None as default, not axis=0
-    def test_2dlist(self):
+    def test_2d_list(self):
         #  Test a 2d list
         a = [[10, 20, 30, 40], [50, 60, 70, 80], [90, 100, 110, 120]]
         b = 52.8885199
-        self.do(a, b)
+        self.equal_test(a, b)
 
-    def test_2darray(self):
+    def test_2d_array(self):
         #  Test a 2d array
         a = [[10, 20, 30, 40], [50, 60, 70, 80], [90, 100, 110, 120]]
         b = 52.8885199
-        self.do(np.array(a), b)
+        self.equal_test(array(a), b)
 
-    def test_2dma(self):
+    def test_2d_ma(self):
         #  Test a 2d masked array
         a = [[10, 20, 30, 40], [50, 60, 70, 80], [90, 100, 110, 120]]
         b = 52.8885199
-        self.do(np.ma.array(a), b)
+        self.equal_test(np.ma.array(a), b)
 
-    def test_2daxis0(self):
+    def test_2d_axis0(self):
         #  Test a 2d list with axis=0
         a = [[10, 20, 30, 40], [50, 60, 70, 80], [90, 100, 110, 120]]
         b = np.array([35.56893304, 49.32424149, 61.3579244, 72.68482371])
-        self.do(a, b, axis=0)
+        self.equal_test(a, b, axis=0)
 
-    def test_2daxis1(self):
+        a = array([[1, 2, 3, 4], [1, 2, 3, 4], [1, 2, 3, 4]])
+        desired = array([1, 2, 3, 4])
+        self.equal_test(a, desired, axis=0, decimal=14)
+
+        a = array([[1, 2, 3, 4], [1, 2, 3, 4], [1, 2, 3, 4]])
+        desired = stats.gmean(a, axis=0)
```

Comment:
Does this and the above assertion test for the same thing?

Reply:
At a first time, I also feel that test for the same thing.
but first case is desired value is analytic result by hand.
second case is using `stats.gmean` result.

Reply:
Your correct, but I've just looked over the test code being run, and I think there is another problem in the test:

https://github.com/scipy/scipy/blob/7bccecdedb11dadcfbc8d7d2c71ac64bb2d87a63/scipy/stats/tests/test_stats.py#L3681-L3685

The second test compares ``gmean(a, axis=axis, dtype=dtype)`` and ``desired``, but desired is already calculated as ``gmean(a, axis=0)`` which should evaluate to the same value everytime everytime, regardless of whether ``gmean`` actually finds the correct value.